### PR TITLE
feat(ui-test): connect flow

### PR DESCRIPTION
**Pull request by arein**
# Breaking Changes

N/A

# Changes

- feat(ui-test): covers the simple connection flow

# Associated Issues

closes #...

### CODE REVIEW

**Review by chris13524 — APPROVED** [review]
IMO too much OOP when functional would be easier to follow, but lgtm

**Review by tomiir — DISMISSED** [review]

File: apps/laboratory/tests/shared/fixtures/w3m-fixture.ts [hunk 1]
```diff
@@ -12,6 +14,10 @@ export const testM = base.extend<ModalFixture>({
     const modalPage = new ModalPage(page)
     await modalPage.load()
     await use(modalPage)
+  },
+  modalValidator: async ({ modalPage }, use) => {
+    const modalValidator = new ModalValidator(modalPage.page)
+    await use(modalValidator)
   }
 })
 export { expect } from '@playwright/test'
```
Comment (0xAsimetriq) [inline apps/laboratory/tests/shared/fixtures/w3m-fixture.ts:23]: Any reason to export `expect` here vs importing it where it is needed, is this part of playwright's api or something similar?

**Review by svenvoskamp — APPROVED** [review]

File: apps/laboratory/tests/shared/pages/WalletPage.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  /**
+   * Handle a session proposal event in the wallet
+   * @param reqAccounts - required account numbers to select ex/ ['1', '2']
+   * @param optAccounts - optional account numbers to select ex/ ['1', '2']
+   * @param accept - accept or reject the session
+   */
+  async handleSessionProposal(opts: SessionParams) {
+    const meh = opts.accept ? `approve` : `reject`
```
Comment (0xAsimetriq) [inline apps/laboratory/tests/shared/pages/WalletPage.ts:49]: Maybe `variant` and not meh 😅

File: apps/laboratory/tests/shared/fixtures/w3m-fixture.ts (hunk 1 above)
Comment (arein) [reply apps/laboratory/tests/shared/fixtures/w3m-fixture.ts:23]: I copied the code from `V2`. I will assess your feedback and refactor in the next iteration. Multiple patterns here that I think can be improved.

I expect this test suite to become very large so important that we have a good setup

**Review by tomiir — APPROVED** [review]

**Pull request merged**